lib/naughty/core.lua

```diff
@@ -420,159 +318,178 @@ function naughty.getById(id)
-    -- Try to set the text while only interpreting <br>.
-    if not setMarkup("<br.->", "\n") then
-        -- That failed, escape everything which might cause an error from pango
-        if not setMarkup(escape_pattern, escape_subs) then
-            -- Ok, just ignore all pango markup. If this fails, we got some invalid utf8
-            if not pcall(setText) then
-                textbox:set_markup("<i>&lt;Invalid markup or UTF8, cannot display message&gt;</i>")
-            end
-        end
+    if self.destroy_cb and reason ~= naughty.notification_closed_reason.silent then
+        self.destroy_cb(reason or naughty.notification_closed_reason.undefined)
     end
 end
 
-local function update_size(notification)
+naughty.connect_signal("destroyed", cleanup)
 
-    local n = notification
-    local s = n.size_info
-    local width = s.width
-    local height = s.height
-    local margin = s.margin
+local function get_screen(s)
+    return s and capi.screen[s]
+end
 
-    -- calculate the width
-    if not width then
-        local w, _ = n.textbox:get_preferred_size(n.screen)
-        width = w + (n.iconbox and s.icon_w + 2 * margin or 0) + 2 * margin
-    end
+-- Proxy the global suspension state on all notification objects
+local function get_suspended(self)
+    return properties.suspended and not self.ignore_suspend
+end
 
-    if width < s.actions_max_width then
-        width = s.actions_max_width
-    end
+function naughty.set_expiration_paused(p)
+    properties.expiration_paused = p
 
-    if s.max_width then
-        width = math.min(width, s.max_width)
+    if not p then
+         for _, n in ipairs(naughty.notifications._expired[1]) do
+            n:destroy(naughty.notification_closed_reason.expired)
+         end
     end
+end
 
-    -- calculate the height
-    if not height then
-        local w = width - (n.iconbox and s.icon_w + 2 * margin or 0) - 2 * margin
-        local h = n.textbox:get_height_for_width(w, n.screen)
-        if n.iconbox and s.icon_h + 2 * margin > h + 2 * margin then
-            height = s.icon_h + 2 * margin
-        else
-            height = h + 2 * margin
-        end
-    end
+--- Emitted when a notification is created.
+-- @signal added
+-- @tparam naughty.notification notification The notification object
 
-    height = height + s.actions_total_height
+--- Emitted when a notification is destroyed.
+-- @signal destroyed
+-- @tparam naughty.notification notification The notification object
 
-    if s.max_height then
-        height = math.min(height, s.max_height)
-    end
+--- Emitted when a notification has to be displayed.
+--
+-- To add an handler, use:
+--
+--    naughty.connect_signal("request::display", function(notification, args)
+--        -- do something
+--    end)
+--
+-- @tparam table notification The `naughty.notification` object.
+-- @tparam table args Any arguments passed to the `naughty.notify` function,
+--  including, but not limited to all `naughty.notification` properties.
```

Comment:
`, but not limited to, all` <-- add comma between to and all

Reply:
Done, thanks